gnovm/pkg/gnolang/preprocess.go

```diff
@@ -2490,6 +2490,22 @@ func checkOrConvertType(store Store, last BlockNode, x *Expr, t Type, autoNative
 	}
 }
 
+func checkWithoutConvertType(store Store, last BlockNode, x *Expr, t Type, autoNative bool) {
```

Comment:
I found there is a similar "optimised case" -- `checkOrConvertIntegerType`. What do you think of changing this function to something like this?

```go
// like checkOrConvertType() but for bools (ie. for and if).
func checkOrConvertIntegerType(store Store, last BlockNode, x Expr) {
	if cx, ok := x.(*ConstExpr); ok {
		convertConst(store, last, cx, BoolType)
	} else if x != nil {
		xt := evalStaticTypeOf(store, last, x)
		if xt.Kind() != BoolKind { /* panic... */ }
	}
}
```

I think this might push the performance improvement a bit further :)